file_organizer_v2/src/file_organizer/cli/undo_redo.py

```
@@ -0,0 +1,352 @@
+            if operation_id:
+                can_undo, reason = manager.can_undo(operation_id)
+                if can_undo:
+                    operations = [op for op in manager.get_undo_stack() if op.id == operation_id]
+                    if operations:
+                        op = operations[0]
+                        print(f"\nWould undo operation {operation_id}:")
+                        print(f"  Type: {op.operation_type.value}")
+                        print(f"  Source: {op.source_path}")
+                        if op.destination_path:
+                            print(f"  Destination: {op.destination_path}")
+                        print(f"\n✓ This operation can be safely undone")
```

Ruff (0.14.13)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)